### ISSUE TITLE
New users receive "Unable to verify your data submission"

### ISSUE THREAD

**Issue by martinbeek** [humhub/humhub#2485]
Three new users who tried to register received this error yesterday and today. This happens when they've chosen a password and entered their name and pressed the continue button.

This is logged in the humhub log:

`exception 'yii\web\BadRequestHttpException' with message 'Unable to verify your data submission.' in /var/www/bsmug.nl/httpdocs/protected/vendor/yiisoft/yii2/web/Controller.php:166 Stack trace: #0 /var/www/bsmug.nl/httpdocs/protected/humhub/components/Controller.php(146): yii\web\Controller->beforeAction(Object(yii\base\InlineAction)) #1 /var/www/bsmug.nl/httpdocs/protected/humhub/modules/user/controllers/AuthController.php(58): humhub\components\Controller->beforeAction(Object(yii\base\InlineAction)) #2 /var/www/bsmug.nl/httpdocs/protected/vendor/yiisoft/yii2/base/Controller.php(154): humhub\modules\user\controllers\AuthController->beforeAction(Object(yii\base\InlineAction)) #3 /var/www/bsmug.nl/httpdocs/protected/vendor/yiisoft/yii2/base/Module.php(523): yii\base\Controller->runAction('login', Array) #4 /var/www/bsmug.nl/httpdocs/protected/vendor/yiisoft/yii2/web/Application.php(102): yii\base\Module->runAction('user/auth/login', Array) #5 /var/www/bsmug.nl/httpdocs/protected/vendor/yiisoft/yii2/base/Application.php(380): yii\web\Application->handleRequest(Object(humhub\components\Request)) #6 /var/www/bsmug.nl/httpdocs/index.php(24): yii\base\Application->run() #7 {main}` 

I've seen it several times before the last few weeks. 

I hope there is a solution, because there are many more errors in the log, from people who have tried to register and did not report the error and just moved on. 

Thanks
Martin

**Comment by acs-ferreira** [comment]
I'm not sure if this can help, but can you try this PR: https://github.com/humhub/humhub/pull/2486

**Comment by martinbeek** [comment]
Thank you. 
I've removed the csrfMetaTags load from the main.php in my themes. Let's see how it goes! 

Martin

**Comment by acs-ferreira** [comment]
Verify that a token is present in the source code on your browser on the affected page.

**Comment by martinbeek** [comment]
I've checked, by subscribing as new user, and the tokens are there on every page. Once now, instead of double as before. 

Let's hope this helps...

**Comment by acs-ferreira** [comment]
Any news on this?

**Comment by martinbeek** [comment]
Yep. We have a new Facebook promo installed, attracting new users and the errors are coming in again. From people trying to register an account. Approx. one out of ten gives this error. Since the people leave no feedback but quickly move on, we have no idea what a user actually sees.

**Comment by martinbeek** [comment]
In Google analytics i can see that the new visitors' page title has the name Error in it. That's all.

**Comment by martinbeek** [comment]
I think i've solved this! 
I removed the default user group with ID number 2. 
I added a new user group "members" which got ID 4. In admin->user settings i assigned this new group to new users. BUT all new users still got group id 2! 
I changed the group ID of my new group to 2 and made all users member of this group.
So it seems that the registration mechanism requires a group with ID 2...

Martin
